features/storage/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -466,6 +466,30 @@ void test_program_read_small_data_sizes()
     delete block_device;
 }
 
+void test_get_type_functionality()
+{
+    BlockDevice *block_device = BlockDevice::get_default_instance();
+    if (block_device == NULL) {
+        TEST_SKIP_MESSAGE("No block device component is defined for this target");
+        return;
+    }
+    const char * bd_type = block_device->get_type();
+    TEST_ASSERT_NOT_EQUAL(0, bd_type);
```

Comment:
Your example, in design docs, suggest NULL is legal value, here you assert on NULL it may be confusing to people.

Reply:
NULL may be valid if there is no underlying bd and the blockdevice yet. However its not valid for the test and therefore assert on NULL.